### DISCUSSION TITLE
Pixi v6 global namespace

### DISCUSSION THREAD

**Discussion by goldenratio** [pixijs/pixijs#8294] [Q&A]
Hi all,

We have our code base (on Pixi v5), where we don't use named imports. 
Example:
```
const app = new PIXI.Application({
  width: 600,
  height: 600,
  backgroundColor: 0x2c3e50
});
```

we also don't have `import * as PIXI from 'pixi.js-legacy';` in each file. This is valid and works(compiles) in Pixi v5.

But when upgrading to v6, existing code doesn't compile. This due to pixi.js-legacy type definitions doesn't use the global namespace.

Do we have workaround or solution for it? So I don't have to change our code to use named imports

**Comment by miltoncandelero** [comment]
I need to know a bit more about your stack.
As I understand you are using the PIXI object so you are not using a bundler? (for example, webpack)

Maybe you have the wrong import in your index.html? you might be importing the `es` browser bundle instead of the old `iife` or just `browser`  module?

**Reply by goldenratio** [reply]
@miltoncandelero 
I have created a minimal repo,

pixi v5 - https://github.com/goldenratio/pixiv6-import-demo
Everything works as expected.

pixi v6 - https://github.com/goldenratio/pixiv6-import-demo/tree/pixiv6
I am getting type errors,
![image](https://user-images.githubusercontent.com/3019293/164089971-12cfc361-c23d-49ac-bf57-acd42504baca.png)

My goal in Pixi v6 branch, is not to make any changes to the imports.

To run the project,
```
npm i
npm run build:watch
```

I am using webpack's expose-loader, to globally expose PIXI https://github.com/goldenratio/pixiv6-import-demo/blob/main/webpack.config.js#L44

**Reply by miltoncandelero** [reply]
I use named imports so I am not very well versed in `expose-loader` but a quick read on the docs: https://webpack.js.org/loaders/expose-loader/#using-configuration leads me to believe that you have the wrong config

`test: path.resolve('./node_modules/pixi.js-legacy'),` on line 44  should be `test: require.resolve('pixi.js-legacy'),`
